core/src/main/java/org/apache/gravitino/listener/TagEventDispatcher.java

```diff
@@ -201,8 +250,12 @@ public String[] associateTagsForMetadataObject(
   public Tag getTagForMetadataObject(String metalake, MetadataObject metadataObject, String name) {
     // TODO: getTagForMetadataObjectPreEvent
     try {
-      // TODO: getTagForMetadataObjectEvent
-      return dispatcher.getTagForMetadataObject(metalake, metadataObject, name);
+      Tag tag = dispatcher.getTagForMetadataObject(metalake, metadataObject, name);
+      TagInfo tagInfo = new TagInfo(tag.name(), tag.comment(), tag.properties());
+      eventBus.dispatchEvent(
+          new GetTagForMetadataObjectEvent(
+              PrincipalUtils.getCurrentUserName(), metalake, metadataObject, name, tagInfo));
```

Comment:
remove `name` from `GetTagForMetadataObjectEvent`, since the tag name is included in tagInfo